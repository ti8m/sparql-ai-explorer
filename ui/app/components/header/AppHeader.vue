<script setup>
import logo_lindas from "~/assets/logo_lindas.png";

const {$version} = useNuxtApp();
</script>

<template>
  <header class="bg-blue-whale p-3 flex items-center justify-between">
    <div class="flex items-center gap-4">
      <a href="/" class="flex items-end">
        <!--<img
          :src="logo_lindas"
          alt="Parlamentvsdienste Icon"
          class="h-12 mx-2"
        />-->
        <h1 class="text-white mb-0">LINDAS SPARQL UI</h1>
      </a>
    </div>
    <p v-if="$version" class="text-white hidden sm:block">v{{ $version }}</p>
  </header>
</template>
