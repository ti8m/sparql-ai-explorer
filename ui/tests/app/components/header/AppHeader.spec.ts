import {mountSuspended} from "@nuxt/test-utils/runtime";
import AppHeader from "~/components/header/AppHeader.vue";
import {it, expect, describe, beforeEach} from "vitest";

import type {Props, TestComponent} from "~~/tests/types/types";

describe("AppHeader", () => {
    let component: TestComponent;

    const factory = async (props: Props) => {
        return await mountSuspended(AppHeader, {
            props: {
                ...props,
            },
        });
    };

    beforeEach(async () => {
        component = await factory({});
    });

    it("should match html", () => {
        expect(component.html()).toMatchSnapshot();
    });
});
