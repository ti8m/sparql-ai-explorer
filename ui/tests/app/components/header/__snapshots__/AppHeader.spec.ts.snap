// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppHeader > should match html 1`] = `
"<header class="bg-blue-whale p-3 flex items-center justify-between">
  <div class="flex items-center gap-4"><a href="/"><img src="/assets/icon_parlament.svg" alt="Parlamentvsdienste Icon" class="h-full mx-2"></a>
    <p class="text-white">The Federal Assembly â€” The Swiss Parliament</p>
  </div>
  <p class="text-white hidden sm:block">v1.0.0</p>
</header>"
`;
